src/unittest/python/tracker_tests.py

```diff
@@ -0,0 +1,73 @@
+from unittest import TestCase, mock
+
+from vyu import tracker
+
+
+class TestTrackerWaitForFixation(TestCase):
+
+    def setUp(self):
+        self.reader_patch = mock.patch('vyu.tracker.imageio.get_reader')
+        self.timer_patch = mock.patch('vyu.tracker.Timer')
+        self.i2pos_patch = mock.patch('vyu.tracker.image2position')
+        self.frames = [mock.Mock(), mock.Mock()]
+
+        self.mock_get_reader = self.reader_patch.start()
+        self.mock_timer = self.timer_patch.start()
+        self.mock_i2pos = self.i2pos_patch.start()
+
+        self.mock_get_reader.return_value.__iter__.return_value = self.frames
```

Comment:
I'm not sure what is being done in this line and I don't know what the use of "__iter__" is here.

Reply:
This is mocking the iter functionality: We call `get_reader` to get a reader for the frames from the webcam. Then we use that reader in the following way:
```python
reader = get_reader()
for frame in reader:
    # Do something here
```
The `for frame in reader` line takes the reader and calls the `__iter__` function on it. It then takes that element and iterates over it by calling `next` everytime it gets back to the `for` statement. If we want to mock this process, we have to do two things:
1. We mock the `get_reader` function,
2. We set the return value of the `__iter__` method of the returned reader to contain the frames we want to inject.
That's wahat this line is doing.